analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Mutex.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                m4.WaitOne(); // FN
+            }
+
+            var m5 = new Mutex(false);
+            var acquired = m5.WaitOne(200, true); // FN, not released on the corect flow
+            if (acquired)
+            {
+                // here it should be released
+            }
+            else
+            {
+                m5.ReleaseMutex(); // this is a programming error
+            }
+
+            var isAcquired = paramMutex.WaitOne(400, false); // FN
+            if (isAcquired)
+            {
+                // not released
+            }
+        }
+
+        public static void UnsupportedWaitAny(Mutex m1, Mutex m2, Mutex m3)
+        {
+            // it is too complex to support this scenario
+            WaitHandle[] handles = new[] { m1, m2, m3 };
+            var index = WaitHandle.WaitAny(handles);
+            // the mutex at the given index should be released
+        }
+
+        public static void UnsupportedWaitAll(Mutex m1, Mutex m2, Mutex m3)
+        {
+            // it is too complex to support this scenario
+            WaitHandle[] handles = new[] { m1, m2, m3 };
+            var allHaveBeenAcquired = WaitHandle.WaitAll(handles);
+            if (allHaveBeenAcquired)
+            {
+                // all indexes should be released
+            }
+        }
+
+        public static void CompliantNotAcquired(Mutex paramMutex)
+        {
+            var m1 = new Mutex(false);
+            var m2 = Mutex.OpenExisting("foo");
+            if (Mutex.TryOpenExisting("foo", out var m3))
+            {
+                // do stuff but don't acquire
+            }
+            var m4 = new Mutex(initiallyOwned: false, "foo", out var mutexWasCreated);
+            if (paramMutex != null)
+            {
+                // do stuff but don't acquire
+            }
+        }
+
+        public static void CompliantAcquiredAndReleased(Mutex paramMutex)
+        {
+            var m1 = new Mutex(false);
+            m1.WaitOne();
+            m1.ReleaseMutex();
+
+            var m2 = Mutex.OpenExisting("foo");
+            if (m2.WaitOne(500))
+            {
+                m2.ReleaseMutex();
+            }
+
+            var isAcquired = paramMutex.WaitOne(400, false);
+            if (isAcquired)
+            {
+                paramMutex.ReleaseMutex();
+            }
+        }
+
+        public static void CompliantComplex(string mutexName, bool shouldAcquire)
+        {
+            Mutex m = null;
+            bool acquired = false;
+            try
+            {
+                m = Mutex.OpenExisting(mutexName);
+                if (shouldAcquire)
+                {
+                    m.WaitOne();
+                    acquired = true;
+                }
+            }
+            catch (UnauthorizedAccessException)
+            {
+                return;
+            }
+            finally
+            {
+                // there can be other exceptions when opening
```

Comment:
I'd remove the comment. Even the `UnauthorizedAccessException` will make it to this `finally`.
```suggestion
```